plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -804,6 +805,47 @@ public void testSqlServerDatetimeOffset(ZoneId sessionZone)
                 .execute(getQueryRunner(), session, sqlServerCreateAndInsert("test_sqlserver_datetimeoffset"));
     }
 
+    @Test
+    public void testSqlServerUniqueIdentifier()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("uniqueidentifier", "NULL", UUID, "CAST(NULL AS uuid)")
+                .addRoundTrip("uniqueidentifier", "'d5416a8a-5faf-208d-a0cc-aaf97732f49a'", UUID, "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'")
+                .addRoundTrip("uniqueidentifier", "'00000000-0000-0000-0000-000000000000'", UUID, "uuid '00000000-0000-0000-0000-000000000000'")
+                .addRoundTrip("uniqueidentifier", "'ffffffff-ffff-ffff-ffff-ffffffffffff'", UUID, "uuid 'ffffffff-ffff-ffff-ffff-ffffffffffff'")
+                .addRoundTrip("uniqueidentifier", "'0E984725-C51C-4BF4-9960-E1C80E27ABA0wrong'", UUID, "uuid '0E984725-C51C-4BF4-9960-E1C80E27ABA0'")
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_sqlserver_uniqueidentifier"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("uniqueidentifier", "NULL", UUID, "CAST(NULL AS uuid)")
+                .addRoundTrip("uniqueidentifier", "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'", UUID, "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'")
+                .addRoundTrip("uniqueidentifier", "uuid '00000000-0000-0000-0000-000000000000'", UUID, "uuid '00000000-0000-0000-0000-000000000000'")
+                .addRoundTrip("uniqueidentifier", "uuid 'ffffffff-ffff-ffff-ffff-ffffffffffff'", UUID, "uuid 'ffffffff-ffff-ffff-ffff-ffffffffffff'")
+                .execute(getQueryRunner(), sqlServerCreateAndTrinoInsert("test_sqlserver_uniqueidentifier"));
+    }
+
+    @Test
+    public void testUnsupportedUniqueIdentifier()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_unsupported_uniqueidentifier", "(data uniqueidentifier)")) {
+            assertSqlServerQueryFails(
+                    "INSERT INTO " + table.getName() + " VALUES ('0E984725-EXTRA-4BF4-9960-E1C80E27ABA0')",
+                    "Conversion failed when converting from a character string to uniqueidentifier.");
+        }
+    }
+
+    @Test
+    public void testTrinoUuid()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("uuid", "NULL", UUID, "CAST(NULL AS uuid)")
+                .addRoundTrip("uuid", "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'", UUID, "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'")
+                .addRoundTrip("uuid", "uuid '00000000-0000-0000-0000-000000000000'", UUID, "uuid '00000000-0000-0000-0000-000000000000'")
+                .addRoundTrip("uuid", "uuid 'ffffffff-ffff-ffff-ffff-ffffffffffff'", UUID, "uuid 'ffffffff-ffff-ffff-ffff-ffffffffffff'")
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_uuid"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_uuid"));
```

Comment:
Keep the code style consistent by putting `trinoCreateAsSelect ` before `trinoCreateAndInsert`.

Most of the connectors follow this rule, except for some cases in `TestCassandraTypeMapping` and `BaseSqlServerTypeMapping`, and we feel we can fix them as a follow-up.
(see output of `git grep -na1 'trinoCreateAsSelect('`)

Does this make sense to you? @hashhar @ebyhr

Reply:
Forcing method order among connectors is overkill in my opinion.